### PR TITLE
Module Three Capstone Project Submission

### PR DESCRIPTION
**Pull request by AbdimulhinYussuf3675**
### **Highlights**

- [x]  Linters added
- [x]  Mobile First Approach
- [x] Api Used
- [x] Redux Store used
- [x] Testing added

### CODE REVIEW

File: src/components/routes/Home.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          <HomeIcon className="svg_icons" />
+        </div>
+        <div className="search">
+          {' '}
+          <input
+            type="text"
+            name="searchItem"
+            placeholder="search for a country"
+            value={searchItem}
+            onChange={handleSearch}
```
Comment (BregornOriginal) [inline src/components/routes/Home.js:43]: Good work creating a search bar 💯 , but I believe that you misunderstand the requirements of this project

There's no need to create a search bar, the requirements say that you need to filter the list of objects from your API by some property, like `name` `Population` `Continent` `Currency`, and other applicable properties from the objects

- Kindly create the necessary filter to list items by name or another property from your objects

File: README.md [hunk 2]
```diff
@@ -43,15 +43,17 @@
...
 - **Redux State Management**
 
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
 
 ## 🚀 Live Demo <a name="live-demo"></a>
 
-- [Live Demo](https://adamspace.onrender.com/).
+- [Live Demo](https://adam-census.onrender.com/)
+
+- [Project Presentation](https://www.loom.com/share/495c60e80cbf433a81e7bf0b095a516c).
```
Comment (BregornOriginal) [inline README.md:56]: - Kindly somewhere in your Readme file, maybe better in the Acknolowdge section, give appropriate credit to the author [Nelson Sakwa on Behance](https://www.behance.net/sakwadesignstudio)

**Review by BregornOriginal — CHANGES_REQUESTED** [review]
## Hi @AbdimulhinYussuf3675  

## Good job so far!
## There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

one-step-closer-one-more

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/components/routes/Home.js (hunk 1 above)
Comment (AbdimulhinYussuf3675) [reply src/components/routes/Home.js:43]: Hello @BregornOriginal, Thanks for the review. On the filter issue, I have filtered using the property `name` as per the requirement. Kindly check lines 20 to 23.

File: src/components/routes/Home.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+          />
```
Comment (rustam-tolipov) [inline src/components/routes/Home.js:44]: **👔 Excellent implementation in response to the previous reviewer's request.**

- [ ] 👨‍💻 However, as previously mentioned by the reviewer, it is necessary to use a filter for loading data. While it is noted that you have implemented filtering for searching, it is also important to filter the main fetched data for loading purposes.

🙏 We kindly request that you take into account the previous reviewer's feedback and make the necessary adjustments. Thank you!

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hi @AbdimulhinYussuf3675, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
1. Great design 💯 
2. Descriptive readme 💯 
3. Kudos for adding searchbar 🥇 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/routes/Home.js (hunk 3 above)
Comment (AbdimulhinYussuf3675) [reply src/components/routes/Home.js:44]: Hello @Rustamxon7 can you explain to me further? I don't get the changes required

Comment (AbdimulhinYussuf3675) [reply src/components/routes/Home.js:44]: Hello can i get a feedback atleast

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @AbdimulhinYussuf3675 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Awesome work on the searching (filtering) functionality. You've done the [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=When%20the%20page%20loads%2C%20the%20webapp%20shows%20the%20list%20of%20categories%20that%20could%20be%20filtered%20out%20by%20some%20parameter%2C%20for%20example%20by%20the%20category%20name.) correctly
- Thank you for communicating through the comments. ❤️ 🙏

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**